### DISCUSSION TITLE
custom oversampling

### DISCUSSION THREAD

**Discussion by rpatel3001** [szpajder/dumpvdl2#26] [Q&A] [answered]
For hardware devices:

>Note: The device must support a sampling rate of 2100000 samples per second to work correctly with dumpvdl2. It is therefore not possible to use devices which only support predefined, fixed sampling rates (notably Airspies). This limitation will be removed in a future release of dumpvdl2.

For recorded IQ files:

> The symbol rate for VDL2 is 10500 symbols/sec. dumpvdl2 internal processing rate is 10 samples per symbol. Therefore the file must be recorded with sampling rate set to an integer multiple of 105000. Specify the multiplier value with --oversample option. The default value is 10, which is valid for files sampled as 1050000 samples/sec.

Is there a technical limitation for not allowing --oversample for rtl_sdr, soapysdr, etc? I'm sharing a receiver between dumpvdl2 and rtlsdr-airband (http://github.com/rpatel3001/docker-rtlmuxer) and would like to up the sample rate to a higher multiple of 105000 to squeeze in a few more audio channels. 

I might try it out myself but it might take some time to get around to it, asking here to see if there's a particular reason it won;t work.

**Comment by szpajder** [comment, marked as answer]
> Is there a technical limitation for not allowing --oversample for rtl_sdr, soapysdr, etc?

There isn't. It's been hardcoded to a value which allows the radio to cover all VDL2 channels and to make things easier for beginners (ie. reduce the number of parameter that might be set incorrectly).

**Reply by rpatel3001** [reply]
Got it, thanks. When I get around to this I'll open a PR.

**Reply by rpatel3001** [reply]
@szpajder It seems that this already works perfectly fine, it just isn't reflected by the documentation. 

edit: the demodulator works fine with a different sampling rate, but the hardware is not initialized with the correct rate. this meets my needs so i won't be opening a PR
